public/app/plugins/datasource/elasticsearch/datasource.ts

```diff
@@ -804,36 +789,53 @@ export class ElasticDatasource
     return lastValueFrom(this.getFields());
   }
 
-  getTagValues(options: any) {
+  getTagValues(options: { key: string }) {
     const range = this.timeSrv.timeRange();
     return lastValueFrom(this.getTerms({ field: options.key }, range));
   }
 
-  targetContainsTemplate(target: any) {
+  targetContainsTemplate(target: ElasticsearchQuery) {
     if (this.templateSrv.containsTemplate(target.query) || this.templateSrv.containsTemplate(target.alias)) {
       return true;
     }
 
-    for (const bucketAgg of target.bucketAggs) {
-      if (this.templateSrv.containsTemplate(bucketAgg.field) || this.objectContainsTemplate(bucketAgg.settings)) {
```

Comment:
None of the types in the definition had a `field` field. Could this be for legacy compat? Otherwise it was incorrect.

https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/elasticsearch/dataquery.gen.ts#L15

Reply:
<img width="655" alt="image" src="https://github.com/grafana/grafana/assets/30407135/a1128cd7-7d17-431d-99ba-61c48f3aae2f">

Reply:
A lot of them have fields - check for the ones that `extends BucketAggregationWithField`

Reply:
It's unlucky that this doesn't show inherited properties 😕 

![imagen](https://github.com/grafana/grafana/assets/1069378/a7a62a67-24b2-47f9-abf3-2f39c2efb496)

Reply:
Fixed!